packages/webapp/webapp_server.js

```diff
@@ -815,18 +816,47 @@ var runWebAppServer = function () {
     var localPort = WebAppInternals.parsePort(process.env.PORT) || 0;
     var host = process.env.BIND_IP;
     var localIp = host || '0.0.0.0';
-    httpServer.listen(localPort, localIp, Meteor.bindEnvironment(function() {
-      if (process.env.METEOR_PRINT_ON_LISTEN)
-        console.log("LISTENING"); // must match run-app.js
-
-      var callbacks = onListeningCallbacks;
-      onListeningCallbacks = null;
-      _.each(callbacks, function (x) { x(); });
-
-    }, function (e) {
-      console.error("Error listening:", e);
-      console.error(e && e.stack);
-    }));
+
+    var startHttpServer = function(listenOptions) {
+      httpServer.listen(listenOptions, Meteor.bindEnvironment(function() {
+        if (process.env.METEOR_PRINT_ON_LISTEN)
+          console.log("LISTENING"); // must match run-app.js
+
+        var callbacks = onListeningCallbacks;
+        onListeningCallbacks = null;
+        _.each(callbacks, function (x) { x(); });
+
+      }, function (e) {
+        console.error("Error listening:", e);
+        console.error(e && e.stack);
+      }));
+    };
+
+    if (typeof localPort == "number") {
+      var listenOptions = { port: localPort, host: host };
+    } else {
+      var socketPath = localPort;
+      var listenOptions = { path: socketPath };
+
+      httpServer.on('error', Meteor.bindEnvironment(function(e) {
+        if (e.code == 'EADDRINUSE') {
+	  var clientSocket = new net.Socket();
+          clientSocket.on('error', Meteor.bindEnvironment(function(e) {
+            if (e.code == 'ECONNREFUSED') {
+              console.log("Deleting stale socket file");
+              fs.unlinkSync(socketPath);
+              startHttpServer(listenOptions);
```

Comment:
Are we completely okay with the `startHttpServer` getting called again here as it will have already been called on line 859 below?